### PR TITLE
Fix Makefile targets

### PR DESCRIPTION
**Pull request by yfei1**
This commit
1. Removed meesages.proto from grpc gateway target list.
2. Fixed a typo in `internal/ipb/synchronizer.pb.go`
3. Explicitly called `go build` instead of make commands from `Dockerfile.cmd`. `open-match-base-build` image does not have gcloud installed so when running `make build-images`, we will see a list of `make: gcloud: Command not found` logs.

Ref: https://pantheon.corp.google.com/cloud-build/builds/8e9026e1-831c-4c25-81d5-23f7bed3f1f4?project=open-match-build
```
make: gcloud: Command not found
#
make: gcloud: Command not found
#
cp api/*.json /go/src/open-match.dev/open-match/build/cmd/swaggerui/static/api/
sed -i 's|https://open-match.dev/api/v.*/|/api/|g' /go/src/open-match.dev/open-match/build/cmd/swaggerui/static/config.json
cp third_party/swaggerui/* /go/src/open-match.dev/open-match/build/cmd/swaggerui/static/
make: gcloud: Command not found
mkdir -p /go/src/open-match.dev/open-match/build/cmd/swaggerui/static/api
make: gcloud: Command not found
#
cp -r examples/demo/static /go/src/open-match.dev/open-match/build/cmd/demo-first-match/static
make: gcloud: Command not found
mkdir -p /go/src/open-match.dev/open-match/build/cmd/demo-first-match/
make: gcloud: Command not found
#
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -672,7 +672,7 @@ pkg/pb/frontend.pb.go: pkg/pb/messages.pb.go
 pkg/pb/matchfunction.pb.go: pkg/pb/messages.pb.go
 pkg/pb/mmlogic.pb.go: pkg/pb/messages.pb.go
 pkg/pb/evaluator.pb.go: pkg/pb/messages.pb.go
-internal/pb/synchronizer.pb.go: pkg/pb/messages.pb.go
+internal/ipb/synchronizer.pb.go: pkg/pb/messages.pb.go
```
Comment (Laremere) [inline Makefile]: This change is good, but because you did too many things in one pr, no approval :(

File: Makefile [hunk 2]
```diff
@@ -196,7 +196,7 @@ else
 	endif
 endif
 
-GOLANG_PROTOS = pkg/pb/backend.pb.go pkg/pb/frontend.pb.go pkg/pb/matchfunction.pb.go pkg/pb/messages.pb.go pkg/pb/mmlogic.pb.go pkg/pb/messages.pb.go pkg/pb/evaluator.pb.go internal/ipb/synchronizer.pb.go pkg/pb/backend.pb.gw.go pkg/pb/frontend.pb.gw.go pkg/pb/matchfunction.pb.gw.go pkg/pb/messages.pb.gw.go pkg/pb/mmlogic.pb.gw.go pkg/pb/evaluator.pb.gw.go
+GOLANG_PROTOS = pkg/pb/backend.pb.go pkg/pb/frontend.pb.go pkg/pb/matchfunction.pb.go pkg/pb/mmlogic.pb.go pkg/pb/messages.pb.go pkg/pb/evaluator.pb.go internal/ipb/synchronizer.pb.go pkg/pb/backend.pb.gw.go pkg/pb/frontend.pb.gw.go pkg/pb/matchfunction.pb.gw.go pkg/pb/mmlogic.pb.gw.go pkg/pb/evaluator.pb.gw.go
```
Comment (Laremere) [inline Makefile]: Already done in #725

File: Dockerfile.cmd [hunk 3]
```diff
@@ -18,7 +18,8 @@ WORKDIR /go/src/open-match.dev/open-match
 
 ARG IMAGE_TITLE
 
-RUN make "build/cmd/${IMAGE_TITLE}"
+RUN mkdir -p build/cmd/${IMAGE_TITLE}
```
Comment (Laremere) [inline Dockerfile.cmd]: The make command doesn't only run go build.  It also takes care of copying and other arbitrary build actions.

Comment (Laremere) [reply Dockerfile.cmd]: Having make's way of building things an docker's way of building things is an intentional unification.  Don't revert it.

Comment (yfei1) [reply Dockerfile.cmd]: Our Makefile has a bunch of third party dependencies. If you run `make build-images` you will be able to see the error message that I mentioned.

Comment (Laremere) [reply Dockerfile.cmd]: I agree, but they're warnings that don't break anything, so I've been ignoring them and haven't prioritized fixing them.

**Pull request merged**